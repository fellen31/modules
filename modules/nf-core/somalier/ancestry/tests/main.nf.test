nextflow_process {

    name "Test Process SOMALIER_ANCESTRY"
    script "../main.nf"
    process "SOMALIER_ANCESTRY"

    tag "modules"
    tag "modules_nfcore"
    tag "somalier"
    tag "somalier/ancestry"
    tag "untar"

    setup {
        run("UNTAR") {
            script "../../../untar/main.nf"
            config "./nextflow.untar.config"

            process {
                """
                input[0] = [
                    [ id:'1kg_test' ], // meta map
                    file("https://zenodo.org/record/3479773/files/1kg.somalier.tar.gz", checkIfExists: true)
                ]
                """
            }
        }
    }

    test("somalier ancestry") {

        when {
            process {
                """
                input[0] = [
                    [ id: 'test' ], // meta map
                    file(params.modules_testdata_base_path + 'delete_me/somalier/whole_genome.somalier', checkIfExists: true)
                ]
                input[1] = Channel.of([
                    [ id: '1kg_test' ],
                    file(params.modules_testdata_base_path + 'delete_me/somalier/ancestry-labels-1kg.tsv", checkIfExists: true)
                ]).join(UNTAR.out.untar)
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}