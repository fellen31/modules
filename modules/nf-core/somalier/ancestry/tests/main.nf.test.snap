{
    "somalier ancestry": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "html": [
                    
                ],
                "tsv": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-06-03T15:13:06.471864838"
    }
}